# Fixed locking the Channels list in cDisplayChannel, where the lock was still held
# when Flush() was called (cont'd).
#
--- 1/menu.c	2018/05/06 09:30:11	5.2
+++ 1/menu.c	2018/05/27 09:51:56	5.3
@@ -4654,8 +4654,10 @@
   displayChannel = Skins.Current()->DisplayChannel(withInfo);
   positioner = NULL;
   channel = NULL;
-  LOCK_CHANNELS_READ;
-  channel = Channels->GetByNumber(cDevice::CurrentChannel());
+  {
+    LOCK_CHANNELS_READ;
+    channel = Channels->GetByNumber(cDevice::CurrentChannel());
+  }
   ProcessKey(FirstKey);
 }
 
